subsys/net/lib/coap_utils/coap_utils.c

```diff
@@ -17,7 +17,7 @@ LOG_MODULE_REGISTER(coap_utils, CONFIG_COAP_UTILS_LOG_LEVEL);
 #define COAP_MAX_REPLIES 1
 #define COAP_POOL_SLEEP 500
 #define COAP_OPEN_SOCKET_SLEEP 200
-#define COAP_RECEIVE_STACK_SIZE 500
+#define COAP_RECEIVE_STACK_SIZE 1000
```

Comment:
Since the problem only occurs with bsdlib and not with the native stack, please make the stack size conditional:
```
#if defined(BSD_LIBRARY)
#define COAP_RECEIVE_STACK_SIZE 1000
#else
#define COAP_RECEIVE_STACK_SIZE 500
#endif
```
No need to increase RAM footprint for samples using native networking stack.

Reply:
Thanks, that's a good point. I've updated the PR.